app/controllers/Study.scala

```diff
@@ -30,33 +30,34 @@ final class Study(
     apiC: => Api
 ) extends LilaController(env):
 
-  def search(text: String, page: Int) = OpenOrScopedBody(parse.anyContent)(_.Study.Read, _.Web.Mobile):
-    Reasonable(page):
-      WithProxy: proxy ?=>
-        val maxLen =
-          if proxy.isFloodish then 50
-          else if HTTPRequest.isCrawler(req).yes then 80
-          else if ctx.isAnon then 100
-          else 200
-        text.trim.some.filter(_.nonEmpty).filter(_.sizeIs > 2).filter(_.sizeIs < maxLen) match
-          case None =>
-            for
-              pag <- env.study.pager.all(Orders.default, page)
-              _ <- preloadMembers(pag)
-              res <- negotiate(
-                Ok.page(views.study.list.all(pag, Orders.default)),
-                apiStudies(pag)
-              )
-            yield res
-          case Some(clean) =>
-            limit.enumeration.search(rateLimited):
-              env
-                .studySearch(clean.take(100), page)
-                .flatMap: pag =>
-                  negotiate(
-                    Ok.page(views.study.list.search(pag, text)),
-                    apiStudies(pag)
-                  )
+  def search(text: String, order: Order, page: Int) =
```

Comment:
Does this break mobile?

Reply:
Make Order in study/search query parameter and optional:
https://github.com/lichess-org/lila/pull/18427/commits/a81af08f0f16f242dbd819fca07e5cfc9eeda7bd